src/Microsoft.ML.Dnn/DnnCatalog.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Data;
+using Microsoft.ML.Transforms;
+using Microsoft.ML.Transforms.Dnn;
+using static Microsoft.ML.Transforms.DnnEstimator;
+
+namespace Microsoft.ML
+{
+    /// <include file='doc.xml' path='doc/members/member[@name="DnnTransformer"]/*' />
+    public static class DnnCatalog
+    {
+
+        /// <summary>
+        /// Retrain the dnn model on new data.
+        /// </summary>
+        /// <param name="catalog"></param>
+        /// <param name="inputColumnNames"> The names of the model inputs.</param>
+        /// <param name="outputColumnNames">The names of the requested model outputs.</param>
+        /// <param name="labelColumnName">Name of the label column.</param>
+        /// <param name="tensorFlowLabel">Name of the node in TensorFlow graph that is used as label during training in TensorFlow.
+        /// The value of <paramref name="labelColumnName"/> from <see cref="IDataView"/> is fed to this node.</param>
+        /// <param name="optimizationOperation">The name of the optimization operation in the TensorFlow graph.</param>
+        /// <param name="modelPath">Path to model file to retrain.</param>
+        /// <param name="epoch">Number of training iterations.</param>
+        /// <param name="batchSize">Number of samples to use for mini-batch training.</param>
+        /// <param name="lossOperation">The name of the operation in the TensorFlow graph to compute training loss (Optional).</param>
+        /// <param name="metricOperation">The name of the operation in the TensorFlow graph to compute performance metric during training (Optional).</param>
+        /// <param name="learningRateOperation">The name of the operation in the TensorFlow graph which sets optimizer learning rate (Optional).</param>
+        /// <param name="learningRate">Learning rate to use during optimization (Optional).</param>
+        /// <param name="addBatchDimensionInput">Add a batch dimension to the input e.g. input = [224, 224, 3] => [-1, 224, 224, 3].
+        /// This parameter is used to deal with models that have unknown shape but the internal operators in the model require data to have batch dimension as well.</param>
+        /// <param name="dnnFramework"></param>
+        /// <remarks>
+        /// The support for retraining is under preview.
+        /// </remarks>
+        public static DnnEstimator RetrainDnnModel(
```

Comment:
Since there's a DnnFramework parameter to discriminate between TF and Torch in the future, the rest of parameters should be shared across TF and Torch.. 

Are we certain that all those parameters are valid and used in Torch? (For sure the parameter tensorFlowLabel is not applicable for Torch, it should be renamed to dnnModelLabel or similar). But are all the rest applicable for both frameworks?

If not, it'd be better to have segregation between parameters that are valid for one or the other framework like by using config classes such as `TensorFlowOptions `and `TorchOptions` depending on the selected framework.

Another possibility is to have two different methods in the catalog instead of just `RetrainDnnModel()`:

- `RetrainTensorflowModel()`
- `RetrainTorchModel()`

Reply:
I think these parameters will be shared. They  are standard DNN parameters. Options class is internal. I will change tensorFlowLabel to dnnLabel

---
In reply to: [310329135](https://github.com/dotnet/machinelearning/pull/4057#discussion_r310329135) [](ancestors = 310329135)